internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,517 @@
+				AgentSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{"infraenv": "local-cluster"},
+				},
+			},
+		}
+		clusterDeployment.Spec.PullSecretRef = &v1.LocalObjectReference{
+			Name: pullSecret.Name,
+		}
+		clusterDeployment.Spec.ClusterName = r.localClusterName
+		clusterDeployment.Spec.BaseDomain = dns.Spec.BaseDomain
+		return nil
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, &clusterDeployment, mutateFn)
+	if err != nil {
+		return errors.Wrap(err, "could not create or ClusterDeployment")
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) importLocalCluster(ctx context.Context, instance *aiv1beta1.AgentServiceConfig) error {
+
+	clusterVersion := &configv1.ClusterVersion{}
+	namespacedName := types.NamespacedName{
+		Name: "version",
+	}
+	err := r.client.Get(ctx, namespacedName, clusterVersion)
+	if err != nil {
+		return errors.Wrap(err, "unable to find cluster version")
+	}
+
+	kubeConfigSecret, err := r.getSecret(ctx, "openshift-kube-apiserver", "node-kubeconfigs")
+	if err != nil {
+		return errors.Wrap(err, "unable to fetch local cluster kubeconfigs")
+	}
+
+	pullSecret, err := r.getSecret(ctx, "openshift-config", "pull-secret")
+	if err != nil {
+		return errors.Wrap(err, "unable to fetch pull secret")
+	}
+
+	dns := &configv1.DNS{}
+	namespacedName = types.NamespacedName{
+		Name: "cluster",
+	}
+	err = r.client.Get(ctx, namespacedName, dns)
+	if err != nil {
+		return errors.Wrap(err, "could not fetch DNS")
+	}
+
+	proxy := &configv1.Proxy{}
+	namespacedName = types.NamespacedName{
+		Name: "cluster",
+	}
+	err = r.client.Get(ctx, namespacedName, proxy)
+	if err != nil {
+		return errors.Wrap(err, "could not fetch proxy")
+	}
+
+	numberOfControlPlaneNodes := 0
+	nodeList := &v1.NodeList{}
+	err = r.client.List(ctx, nodeList)
+	if err != nil {
+		return errors.Wrap(err, "error while fetching nodes")
+	}
+	for _, node := range nodeList.Items {
+		for nodeLabelKey := range node.Labels {
+			if nodeLabelKey == "node-role.kubernetes.io/control-plane" {
+				numberOfControlPlaneNodes++
+			}
+		}
+	}
+
+	if clusterVersion.Status.Desired.Image == "" {
+		return errors.Wrap(err, "unable to determine desired release image")
+	}
+
+	err = r.createOrUpdateClusterImageSet(ctx, clusterVersion.Status.Desired.Image, instance)
+	if err != nil {
+		return err
+	}
+
+	err = r.createOrUpdateNamespace(ctx, r.localClusterName, instance)
+	if err != nil {
+		return err
+	}
+
+	// Store the kubeconfig data in the local cluster namespace
+	err = r.createOrUpdateSecret(
+		ctx,
+		r.localClusterName,
+		fmt.Sprintf("%s-admin-kubeconfig", r.localClusterName),
+		map[string][]byte{"kubeconfig": kubeConfigSecret.Data["lb-ext.kubeconfig"]},
+		instance)
+	if err != nil {
+		return err
+	}
+
+	// Store the pull secret in the local cluster namespace
+	err = r.createOrUpdateSecret(
+		ctx,
+		r.localClusterName,
+		pullSecret.Name,
+		map[string][]byte{".dockerconfigjson": pullSecret.Data[".dockerconfigjson"]},
```

Comment:
Can we just pass `pullSecret.Data` and not have to know the key is `.dockerconfigjson`?